ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -528,12 +537,16 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
         const saveRealAddressAction$ = this.saveTrigger$.pipe(
             withLatestFrom(this.formErrorMessages$, currentAddress$),
             filter(([_, formErrors, addressToSave]) => {
-                return !formErrors?.length && canSaveRealAddress(addressToSave);
+                const canSave = canSaveRealAddress(addressToSave);
+                if (!canSave) {
+                    this.block.hasValidAddress = false;
+                }
+                return !formErrors?.length && canSave;
```

Comment:
should we set hasValidAddress as false if formErrors list is not empty as well?

Reply:
It is not necessary. 
In case we have enough fields values (`canSave= true`), but they are invalid => 
we set the `this.block.hasValidAddress = false;` earlier (before `this.addressService.verifyAddress`, see line 368).